stacks-signer/src/v0/signer_state.rs

```diff
@@ -304,50 +299,122 @@ impl LocalStateMachine {
             return Ok(());
         };
 
-        if !Self::is_timed_out(tenure_id, db, proposal_config)? {
+        let is_timed_out = if state_machine.active_signer_protocol_version
+            < GLOBAL_SIGNER_STATE_ACTIVATION_VERSION
+        {
+            SortitionStateV1::is_timed_out(tenure_id, db, proposal_config.block_proposal_timeout)
+        } else {
+            SortitionStateV2::is_timed_out(
+                tenure_id,
+                db,
+                eval,
+                client.get_signer_address(),
+                proposal_config.block_proposal_timeout,
+            )
+        }?;
```

Comment:
Same comment here -- a SortitionStateVersion enum would be helpful, because you can perform the check once, and then:

```rust
let is_timed_out = if state_version.uses_global_state() {
  ..
```

and then the constructor for `SortitionState::new` would accept the `state_version` rather than a u64, and it could be a match rather than an if-else.